packages/grafana-ui/src/components/DataSourceSettings/AlertingSettings.tsx

```diff
@@ -29,24 +37,46 @@ export function AlertingSettings<T extends AlertingConfig>({
     [alertmanagerDataSources]
   );
 
+  const onManageAlertsToggle = (manageAlertsEnabled: boolean) => {
+    const jsonData = { ...options.jsonData, manageAlerts: manageAlertsEnabled };
+    if (!manageAlertsEnabled) {
+      if (jsonData.ruler) {
+        delete jsonData.ruler;
+      }
+      setCustomRulerHTTPSettings(false);
+    }
+    onOptionsChange({
+      ...options,
+      jsonData,
+    });
+  };
+
+  const [customRulerHTTPSettings, setCustomRulerHTTPSettings] = useState(Boolean(options.jsonData.ruler?.url));
+  const onCustomRulerURLToggle = (checked: boolean) => {
+    setCustomRulerHTTPSettings(checked);
+    if (!checked) {
+      onOptionsChange({
+        ...options,
+        jsonData: {
+          ...(omit(options.jsonData, 'ruler') as T),
+        },
+        secureJsonData: omit(options.secureJsonData ?? {}, 'rulerBasicAuthPassword'),
+        secureJsonFields: omit(options.secureJsonFields, 'rulerBasicAuthPassword'),
+      });
+    }
+  };
+
   return (
     <>
       <h3 className="page-heading">Alerting</h3>
       <div className="gf-form-group">
```

Comment:
Would be nice to replace this with the form components instead of adding more components that are relying on the `gf-*` classes.